src/handler.ts

```diff
@@ -229,21 +257,76 @@ abstract class AbstractFileHandler implements IMessageFilter {
    * Handle a change to one of the widget titles.
    */
   protected titleChanged(title: Title, args: IChangedArgs<any>): void {
-    let widget = arrays.find(this.widgets,
+    let widget = arrays.find(this._widgets,
       (w, index) => { return w.title === title; });
     if (widget === void 0) {
       return
     }
     if (args.name == 'text') {
-      let model = AbstractFileHandler.modelProperty.get(widget);
+      let model = this._getModel(widget);
       let newPath = this.getNewPath(model.path, args.newValue);
       this.manager.rename(model.path, newPath).then(contents =>
-        AbstractFileHandler.modelProperty.set(widget, contents));
+        this._setModel(widget, contents));
     }
   }
 
-  private _manager: IContentsManager;
+  /**
+   * Get the model for a given widget.
+   */
+  private _getModel(widget: Widget) {
+    return Private.modelProperty.get(widget);
+  }
+
+  /**
+   * Set the model for a widget.
+   */
+  private _setModel(widget: Widget, model: IContentsModel) {
+    Private.modelProperty.set(widget, model);
+  }
+
+  /**
+   * Resolve a given widget.
+   */
+  private _resolveWidget(widget: Widget): Widget {
+    widget = widget || this._activeWidget;
+    if (this._widgets.indexOf(widget) === -1) {
+      return;
+    }
+    return widget;
+  }
+
+  /**
+   * Find a widget given a model.
+   */
+  private _findWidgetByModel(model: IContentsModel): Widget {
+    let path = model.path;
+    let index = arrays.findIndex(this._widgets,
```

Comment:
`return arrays.find(this._widgets, widget => this._getModel(widget).path === model.path)`